### PR TITLE
darktable_ucs_22_helpers.h: Fix crash due to missing alignment.

### PR DESCRIPTION
**Pull request by TurboGit**
I had a crash when starting darktable when entering darkroom 1 out of 3 times. With this patch I cannot reproduce.

### CODE REVIEW

**Comment by TurboGit** [comment]
@jenshannoschwalm @ralfbrown : Does this looks ok to you?

**Review by jenshannoschwalm — COMMENTED** [review]
Right!

**Comment by jenshannoschwalm** [comment]
Good to see all maths problems sorted  out before concentrating on the guidung algo.

File: src/common/darktable_ucs_22_helpers.h [hunk 1]
```diff
@@ -233,14 +233,15 @@ static inline struct dt_iop_order_iccprofile_info_t * D65_adapt_iccprofile(struc
...
+      (dt_iop_order_iccprofile_info_t *)dt_alloc_aligned(sizeof(dt_iop_order_iccprofile_info_t));
 
     // Init a new temp profile by copying the base profile
     memcpy(white_adapted_profile, work_profile, sizeof(dt_iop_order_iccprofile_info_t));
 
     // Multiply the in/out matrices by the chromatic adaptation matrix
-    dt_colormatrix_t input_matrix;
-    dt_colormatrix_t output_matrix;
+    dt_colormatrix_t DT_ALIGNED_ARRAY input_matrix;
+    dt_colormatrix_t DT_ALIGNED_ARRAY output_matrix;
```
Comment (ralfbrown) [inline src/common/darktable_ucs_22_helpers.h:244]: The definition of `dt_colormatrix_t` already includes the DT_ALIGNED_ARRAY, so these don't actually do anything.  But the aligned allocation is needed for any embedded fields in `dt_iop_order_iccprofile_info_t` which need to be aligned.

Comment (TurboGit) [reply src/common/darktable_ucs_22_helpers.h]: Confirmed I cannot reproduce the crash with just the aligned allocation. Thanks!

**Comment by ralfbrown** [comment]
It didn't occur to me earlier, but changing `malloc` to `dt_alloc_aligned` means you'll also need to change the corresponding `free` to `dt_free_align` or the Windows build will crash (its implementation of `free` can't handle aligned allocations).

**Comment by TurboGit** [comment]
> It didn't occur to me earlier, but changing `malloc` to `dt_alloc_aligned` means you'll also need to change the corresponding `free` to `dt_free_align` or the Windows build will crash (its implementation of `free` can't handle aligned allocations).

Indeed, now fixed.

**Pull request merged**